### ISSUE TITLE
OSX tray icon clicks always report icon 'bounds' on focused screen

### ISSUE THREAD

**Issue by russellhaering** [atom/electron#1847]
Using:

* Electron `v0.27.2`
* OSX `10.10.3`

With two displays and a tray icon, the bounds object passed to 'click' event handlers always reports the location of the icon on whichever display has the focused window, not the display on which the click occurred.

To reproduce this, on an OSX machine with two displays:

1. Fire up the [menubar example app](https://github.com/maxogden/menubar/tree/master/example)
2. Focus a window on one display
3. Click the tray icon on the other display

The menu will open under the icon on the wrong display.

I'm able to work around this (with a few probable edge cases) using the following patch to menubar:

```patch
diff --git a/index.js b/index.js
index 137c70a..c9aa7ed 100644
--- a/index.js
+++ b/index.js
@@ -38,6 +38,11 @@ module.exports = function create (opts) {
       if (bounds.x === 0 && bounds.y === 0) {
         var size = electronScreen.getPrimaryDisplay().workAreaSize
         bounds.x = size.width // default to top right
+      } else {
+        var requestedWorkArea = electronScreen.getDisplayNearestPoint(bounds).workArea
+        var actualWorkArea = electronScreen.getDisplayNearestPoint(electronScreen.getCursorScreenPoint()).workArea
+        var xOffset = (requestedWorkArea.x + requestedWorkArea.width) - bounds.x
+        bounds.x = actualWorkArea.x + actualWorkArea.width - xOffset
       }
       showWindow(bounds)
     })
```

**Comment by fabien-d** [comment]
> the bounds object passed to 'click' event handlers always reports the location of the icon on whichever display has the focused window, not the display on which the click occurred.

I can confirm, I'm having the same issue. The workaround above works [from the menubar module].

**Comment by jenslind** [comment]
Can't reproduce this in `0.30.2`. @zcbenz has this been fixed?
Found this in the release notes of `0.30.2`:
> A few fixes to Tray's behavior.

**Comment by zcbenz** [comment]
It should have been fixed by ce042d0, thanks for looking into this.
